airflow/utils/task_group.py

```diff
@@ -94,10 +95,15 @@ def __init__(
             self.used_group_ids: Set[Optional[str]] = set()
             self._parent_group = None
         else:
-            if not isinstance(group_id, str):
-                raise ValueError("group_id must be str")
-            if not group_id:
-                raise ValueError("group_id must not be empty")
+            if prefix_group_id:
```

Comment:
Is it better to always do `validate_group_key` (even if `prefix_group_id` is False)?

Reply:
This is a good question. I think not.

If we do it - that's technically backwards incompatible change because some DAGs that already used spaces might stop working. With prefix, those DAGs would not work anyway so it's not breaking anything 'more'. 

I do not think that there is any limitation in case we do not use group id as prefix in task Id. I think it would work and it has nice properties. Spaces and national characters especially might be much better in the name of the group in the UI. We do not have separate 'display name' there and the Id is displayed in the UI, so I'm 100% sure people will use their national characters there (and spaces).

If we find that it does not work with those characters, i think we should rather fix it than prevent it. 

Now when we talk about it - maybe even the right fix should be to remove all the invalid characters from group I'd rather than prevent it even in prefix case ? 

This one can make it to 2.1.3 but then in 2.2 we could add it as a feature to allow such cases.


WDYT?

Reply:
One issue I see with removal is the possibility that nearly _all_ of the characters could be removed, leaving something meaningless behind.

Reply:
Yeah, Removal is a bad word here. We should SLUGIFY them. We could for example use mozilla slugifier: https://github.com/mozilla/unicode-slugify which can do such magic as converting Chinese to basically - ascii:

```
# Replace non ascii chars with their "best" representation
slugify(u'北京 (capital of China)', only_ascii=True)
# u'bei-jing-capital-of-china'
```

Reply:
Alternatively: We should just allow non-ascii characters here (and in DAG and task ids), and follow the same rules as python identifiers (which _do_ allow unicode word characters) https://docs.python.org/3/reference/lexical_analysis.html?highlight=identifier#identifiers

(There was another issue related to this where someone was asking to add a "comment" or description to a DAG for showing in the UI, because the author wanted Chinese dag ids. Can't find that issue now though.)

Reply:
Ah yeah. Adding optional TaskGroupDescription, same  as TaskDescription would be a pretty neat feature and would solve the problem (though seems like for ids the cat is already out-of-the-bag for Airflow 2 due to potential incompatibilities if someone has Task Group with spaces/national characters.).